test/test_repolib.py

```diff
@@ -273,21 +273,46 @@ def stub_content():
         self.assertEquals(1, update_report.updates())
 
     @patch("subscription_manager.repolib.RepoFile")
-    def test_update_when_not_registered_and_existing_repo(self, mock_file):
+    def test_update_when_repo_not_modified_on_mutable(self, mock_file):
         self._inject_mock_invalid_consumer()
         mock_file = mock_file.return_value
-        mock_file.section.return_value = Repo('x', [('gpgcheck', 'original'), ('gpgkey', 'some_key')])
+        mock_file.section = MagicMock(side_effect=[Repo('x', [('gpgcheck', 'original'), ('gpgkey', 'some_key')]), Repo('x', [('gpgcheck', 'original')])])
 
         def stub_content():
             return [Repo('x', [('gpgcheck', 'new'), ('gpgkey', 'new_key'), ('name', 'test')])]
 
         update_action = RepoUpdateActionCommand()
         update_action.get_unique_content = stub_content
-        update_action.perform()
+        current = update_action.perform()
+        # confirming that the assessed value does change when repo file
+        # is the same as the server value file.
+        self.assertEquals('new', current.repo_updates[0]['gpgcheck'])
 
+        # this is the ending server value file.
         written_repo = mock_file.update.call_args[0][0]
-        self.assertEquals('original', written_repo['gpgcheck'])
-        self.assertEquals('new_key', written_repo['gpgkey'])
+        self.assertEquals('new', written_repo['gpgcheck'])
+        self.assertEquals(None, written_repo['gpgkey'])
+
+    @patch("subscription_manager.repolib.RepoFile")
+    def test_update_when_repo_modified_on_mutable(self, mock_file):
+        self._inject_mock_invalid_consumer()
+        mock_file = mock_file.return_value
+        mock_file.section = MagicMock(side_effect=[Repo('x', [('gpgcheck', 'unoriginal'), ('gpgkey', 'some_key')]), Repo('x', [('gpgcheck', 'original')])])
```

Comment:
Suggestion for making this clearer: define the two repos as vars so you end up with something like:

```python
MagicMock(side_effect=[modified_repo, server_repo])
```